woocommerce/woocommerce
# Add Channels card into Multichannel Marketing page

### All Submissions:


### Changes proposed in this Pull Request:

Closes https://github.com/woocommerce/woocommerce/issues/34906. Uses changes introduced in code refactor PR https://github.com/woocommerce/woocommerce/pull/36540.

In this PR, we add a new Channels card into the Multichannel Marketing page. The card will display:

- A list of registered channels, i.e. extensions that are installed, activated, and integrated with the Multichannel Marketing library. This is retrieved by calling `GET /marketing/channels`.
- A list of recommended channels. This is retrieved by calling `GET /marketing/recommendations?category=channels`. The data is then filtered with `@woocommerce/data/plugin` so that we would only display recommended channels that are not yet installed or not yet activated.

Screenshot of Channels card when we do not have a registered channel yet - a list of recommended channels would be shown for us to install and activate: 

![image](https://user-images.githubusercontent.com/417342/213877830-50076bf1-5787-4a07-896e-3c91e21fa885.png)

Screenshot of Channels card when we have a registered channel (this example uses PR https://github.com/woocommerce/google-listings-and-ads/pull/1730): 

![image](https://user-images.githubusercontent.com/417342/213877728-feb8ba7c-8289-45d8-bb0a-3fc7fbde1322.png)

Demo video of activating a recommended channel in the Channels card:

https://user-images.githubusercontent.com/417342/213878020-d61a35e8-97e8-44a3-b3e6-dbec7bf499e3.mp4

### How to test the changes in this Pull Request:

Pre-condition: 

- Enable WooCommerce Multichannel Marketing experience in WooCommerce Settings > Advanced > Features (`/wp-admin/admin.php?page=wc-settings&tab=advanced&section=features`).

Test instructions:

1. Go to Marketing page at `/wp-admin/admin.php?page=wc-admin&path=%2Fmarketing`.
2. You should see the Channels card at the top of the page.
3. Install and activate a recommended channel. If the channel is integrated with the Multichannel Marketing library, it will show up as a channel in the card. 
    - Currently, only Google Listings and Ads is integrated with the library, see PR https://github.com/woocommerce/google-listings-and-ads/pull/1730.
4. After installing and activating the channel, you should see a "Finish setup" button. Clicking on the button should bring you to the onboarding setup page of the channel. 
5. After completing the onboarding setup for the channel, go back to the Marketing page. The "Finish setup" button should be gone, and there should be a "Manage" button for the channel. Clicking on the "Manage" button should bring you to the settings page of the channel.

### Other information:




### FOR PR REVIEWER ONLY:
